src/buyflow/Buyflow.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react'
+import AgeStep from './AgeStep'
+import EmailStep from './EmailStep'
+import SummaryStep from './SummaryStep'
+
+interface BuyflowProps {
+  productId: ProductIds
+}
+
+export enum ProductIds {
+  devIns = 'dev_ins',
+}
+
+const PRODUCT_IDS_TO_NAMES = {
+  [ProductIds.devIns]: 'Developer Insurance',
+}
+
+const Buyflow: React.FC<BuyflowProps> = (props) => {
+  const [currentStep, setStep] = useState('email')
+  const [collectedData, updateData] = useState({
+    email: '',
+    age: 0,
+  })
+  const getStepCallback = (nextStep: string) => (field: string, value: any) => {
+    updateData({ ...collectedData, [field]: value })
+    setStep(nextStep)
+  }
+  return (
+    <>
+      <h4>Buying {PRODUCT_IDS_TO_NAMES[props.productId]}</h4>
+      {(currentStep === 'email' && <EmailStep cb={getStepCallback('age')} />) ||
```

Comment:
Important reminder:
since `productConfig` and `StepComponent` would be dynamically inferred, we should make sure that we have error boundaries in case there was some error and the entity doesn't exist for some reason